versions/development/SPEC.md

```diff
@@ -2138,44 +2138,280 @@ Now everything inside of `tasks.wdl` must be accessed through the default namesp
 Each namespace may contain namespaces, tasks, and at most one workflow.  The names of the contained namespaces, tasks, and workflow need to be unique within that namespace. For example, one cannot import two workflows while they have the same namespace identifier. Additionally, a workflow and a namespace both named `foo` cannot exist inside a common namespace. Similarly there cannot be a task `foo` in a workflow also named `foo`.
 However, you can import two workflows with different namespace identifiers that have identically named tasks. For example, you can import namespaces `foo` and `bar`, both of which contain a task `baz`, and you can call `foo.baz` and `bar.baz` from the same primary workflow.
 
-# Scope
+# Value Scopes and Reachability
 
-Scopes are defined as:
+## Top Level Scopes
 
-* `workflow {...}` blocks
-* `call` blocks
-* `if(expr) {...}` blocks
-* `scatter(x in y) {...}` blocks
+The top level scopes which can contain value declarations are:
```

Comment:
This is a minor point, but I think where the spec says `declaration` what's actually meant is `definition`. Conventionally, a declaration binds a name in a scope but doesn't provide the name's value, or require that a value is defined. E.g. in C/C++, you put function declarations in header files and function definitions in source (.c/cpp) files. Some languages don't support declarations separate from definitions, and I think WDL falls into this category. Therefore all named values are definitions (as are tasks and workflows).

Reply:
At a quick glance (IOW: he's shooting from the hip and not validating what he says) this is likely an artifact of "The spec matches terminology to how Hermes describes things", which isn't really a great policy going forward

Reply:
I'll happily go along with `definition`.